lib/npm.js

```diff
@@ -247,6 +247,10 @@ class Npm {
       execWorkspaces = true
     }
 
+    if (command.checkDevEngines && !this.global) {
```

Comment:
What's `global` refer to here - a global install of a package?

Why do we skip if `global` is truthy? Might be worthy of a comment, particularly if tests still pass if its removed.

Reply:
Global installs are special in that from some parts of the code they look like a package being installed and in other parts of the code they look like a package you are installing deps into.  The code base is filled with these kinds of checks already.

Reply:
It might be good to add a test that verifies global install of a package ignores devEngines.

Reply:
thanks for this, added a test to check for this!